### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by AyomipoMajek**
Bookstore: connect to API finished according to microverse requirements which include:

In the previous project you developed the add book and remove book features directly in your application. Now you will refactor your code to use the Bookstore API to add and remove books.

Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend
Use either [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls
Fetch your books from Bookstore API on when the component has rendered with useEffect
Refactor your add book and remove book features to persist your changes in the server
Modify your reducer logic to include HTTP requests to the Bookstore API
Styling is not required
Everything should work the same as before

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -1,37 +1,73 @@
...
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
 };
 
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Q2vJ7mN9IENuKzD4CjC7/books');
```
Comment (Munsa1) [inline src/redux/books/bookSlice.js:8]: - [ ] It is a requirement  that Axios is used to make the HTTP requests, kindly do so

File: src/redux/books/bookSlice.js [hunk 2]
```diff
@@ -1,37 +1,73 @@
...
+    method: 'post',
+    headers: { 'Content-type': 'application/json' },
+    body: JSON.stringify(dat),
+  });
+  response.text();
+  return { dat };
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async ({ itemId }) => {
+  const response = await fetch(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Q2vJ7mN9IENuKzD4CjC7/books/${itemId}`,
```
Comment (Munsa1) [inline src/redux/books/bookSlice.js:31]: - [ ] Again here, It is a requirement  that Axios is used to make the HTTP requests, kindly do so

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @AyomipoMajek , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Used `CreateAsyncThunk` in the project :heavy_check_mark: 
Created `fetchBook` and `addBook` action creators :heavy_check_mark:  

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/bookSlice.js (hunk 1 above)
Comment (AyomipoMajek) [reply src/redux/books/bookSlice.js:8]: the instruction said we should:
Use either [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls

File: src/redux/books/bookSlice.js (hunk 2 above)
Comment (AyomipoMajek) [reply src/redux/books/bookSlice.js:31]: the instruction said we should:
Use either [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls

Comment (AyomipoMajek) [reply src/redux/books/bookSlice.js:31]: Kindly take a look at the instructions again

Comment (AyomipoMajek) [reply src/redux/books/bookSlice.js:31]: Screenshot (27)

**Review by m05e5 — APPROVED** [review]
**Hi @AyomipoMajek 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Clean and well-organized code
✔️ No Linters error
✔️ Correct Gitflow

## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**